mcs/class/corlib/System/TimeZoneInfo.WebAssembly.cs

```diff
@@ -5,23 +5,90 @@
 using System.Collections.ObjectModel;
 using System.IO;
 using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
 
 namespace System {
 
 	public partial class TimeZoneInfo {
 
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
+		extern static void mono_timezone_get_local_name (ref string name);
+
+		static string GetMonoWasmLocalName ()
+		{
+			string localName = null;
+			mono_timezone_get_local_name (ref localName);
+			return localName;
+		}
+
 		static TimeZoneInfo CreateLocal ()
 		{
-			return TimeZoneInfo.Utc;
+			// Call GetMonoWasmData and do not throw on error
+			// If there is no stream data returned then the zone information database
+			// was not found and we should return back a UTC value.
+			using (Stream stream = GetMonoWasmData (null, false)) {
+				if (stream == null)
+					return Utc;
+				return BuildFromStream (GetMonoWasmLocalName (), stream);
+			}
 		}
 
 		static TimeZoneInfo FindSystemTimeZoneByIdCore (string id)
 		{
-			throw new NotImplementedException ();
+			using (Stream stream = GetMonoWasmData (id)) {
+			 	return BuildFromStream (id, stream);
+			}
 		}
 
 		static void GetSystemTimeZonesCore (List<TimeZoneInfo> systemTimeZones)
 		{
+			foreach (string name in GetMonoWasmNames ()) {
+				using (Stream stream = GetMonoWasmData (name, false)) {
+					if (stream == null)
+						continue;
+					systemTimeZones.Add (BuildFromStream (name, stream));
+				}
+			}
+		}
+
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
+		extern static IntPtr mono_timezone_get_names (ref int count);
+
+		static ReadOnlyCollection<string> GetMonoWasmNames ()
+		{
+			int count = 0;
+			IntPtr array = mono_timezone_get_names (ref count);
+			if (count > 0)
+			{
+				string [] names = new string [count];
+				for (int i = 0, offset = 0; i < count; i++, offset += IntPtr.Size) {
```

Comment:
Is there a particular reason why we're doing this awkwardly with IntPtr and manually computing offsets, vs the alternative of just ```byte[]* pArray``` and ```pArray[i]```?